components/brave_new_tab_ui/components/default/cryptoDotCom/index.tsx

```diff
@@ -451,35 +421,690 @@ class CryptoDotCom extends React.PureComponent<Props, State> {
+      const quoteAvailable = this.getAvailableBalanceForCurrency(selectedQuote)
+      const pair = this.props.tradingPairs.filter(pair => pair.base === selectedBase && pair.quote === selectedQuote)
+      return <AssetTrade
+        tickerPrices={this.props.tickerPrices}
+        availableBalanceBase={baseAvailable}
+        availableBalanceQuote={quoteAvailable}
+        base={selectedBase}
+        quote={selectedQuote}
+        priceDecimals={pair[0] ? pair[0].price : '0'}
+        quantityDecimals={pair[0] ? pair[0].quantity : '0'}
+        handleBackClick={this.clearAsset}
+      />
+    }
+
+    if (currentAssetView === AssetViews.DEPOSIT) {
+      const depositAddress = this.props.depositAddresses[selectedBase]
+      return <AssetDeposit
+        assetAddress={depositAddress ? depositAddress.address : ''}
+        assetQR={depositAddress ? depositAddress.qr_code : ''}
+        base={selectedBase}
+        handleBackClick={this.clearAsset}
+      />
+    }
+
+    return null
+  }
+
+  renderIndex () {
+    const { selectedBase } = this.state
 
-    if (!showContent) {
+    if (selectedBase) {
+      return this.renderAssetView()
+    } else {
+      return <>
+        {this.renderNav()}
+        {this.renderCurrentView()}
+      </>
+    }
+  }
+
+  renderDisconnectView = () => {
+    return (
+      <DisconnectWrapper>
+        <DisconnectTitle>
+          {getLocale('cryptoDotComWidgetDisconnectTitle')}
+        </DisconnectTitle>
+        <DisconnectCopy>
+          {getLocale('cryptoDotComWidgetDisconnectText')}
+        </DisconnectCopy>
+        <DisconnectButton onClick={this.props.disconnect}>
+          {getLocale('cryptoDotComWidgetDisconnectButton')}
+        </DisconnectButton>
+        <DismissAction onClick={this.props.cancelDisconnect}>
+          {getLocale('cryptoDotComWidgetCancelText')}
+        </DismissAction>
+      </DisconnectWrapper>
+    )
+  }
+
+  render () {
+    if (!this.props.showContent) {
       return this.renderTitleTab()
     }
 
     return (
-      <WidgetWrapper tabIndex={0}>
-        {this.renderTitle()}
-        {this.renderSelectedView()}
-      </WidgetWrapper>
+      <ThemeProvider theme={{
+          secondary: 'rgba(15, 28, 45, 0.7)',
+          primary: '#44B0FF',
+          danger: 'rgba(234, 78, 92, 1)'
+        }}>
+        <WidgetWrapper>
+          {
+            this.props.disconnectInProgress
+            ? this.renderDisconnectView()
+            : <>
+                {this.renderTitle()}
+                {(this.props.isConnected) ? (
+                  this.renderIndex()
+                ) : (
+                  this.renderPreOptIn()
+                )}
+              </>
+          }
+        </WidgetWrapper>
+      </ThemeProvider>
     )
   }
 }
 
 export const CryptoDotComWidget = createWidget(CryptoDotCom)
+
+// Supporting Components
+
+function AssetDeposit ({
+  assetAddress,
+  assetQR,
+  base,
+  handleBackClick
+}: any) {
```

Comment:
We should hardly ever use the `any` type, especially for crypto-related properties so we're sure we're converting numbers correctly.